lib/cn/node_split.c

```diff
@@ -162,6 +162,9 @@ find_inflection_key(
             break;
     }
 
+    log_debug("node (%lu) inflection key kvlen: %lu/%lu (%lf%%)", tnode->tn_nodeid, kvlen,
+        total_kvlen, (double)kvlen / total_kvlen);
+
```

Comment:
No need for parens around node ID.
I would put all the args on the second line, makes it easier to eyeball them with their percent escapes...
Maybe tag each message with 1/2, pre/post, before/after, or some such notation that pairs them...

Reply:
Why would key count be better? The point is to split the data in half, not the keys in half. Each message is already unique if you read the messages, "inflection key", "split key", "inflection kblock".

Reply:
Can we add a "cn_split" keyword to these log messages? Easier to grep..

Reply:
I don't really want to add that since it is the name of a function, and someone might care to only log messages generated by the cn_split() function

Reply:
Can we say "node **split** %lu inflection..."?

Reply:
Something interesting that GLib does for logging is allow you to set a `G_LOG_DOMAIN` which is a keyword prefix for every log statement within that compilation unit. Might be worth coming up with a similar idiom to solve this kind of use case.

https://docs.gtk.org/glib/logging.html#log-domains

Reply:
Why not grep on the function name?

Reply:
All our log message emit "HSE" at the beginning, I think David put that in long long ago...

Reply:
We usually print "node %lu" in most messages, would be good to keep that idiom.  "split" is in the name of the function, right?

Reply:
I put together https://github.com/hse-project/hse/pull/383 to encapsulate my thoughts. I know we put [HSE], which is useful, but I think something in addition to that would be useful because Nabeel for instance just wants to see HSE logs related to node split, but there isn't just one function to grep for. This should make it easier to view logs related to the same idea if used well.

Reply:
log_info and family already add function name to log message.

Reply:
That isn't what the proposal is about. Most of the time code for a given domain (ie node split) spans multiple functions.